code/modules/power/cell.dm

```diff
@@ -333,3 +333,17 @@
 
 /obj/item/stock_parts/cell/beam_rifle/emp_act(severity)
 	charge = Clamp((charge-(10000/severity)),0,maxcharge)
+
+/obj/item/stock_parts/cell/emergency_light
+	name = "miniature power cell"
+	desc = "A tiny power cell with a very low power capacity. Used in light fixtures to power them in the event of an outage."
+	maxcharge = 120 //Emergency lights use 0.2 W per tick, meaning ~10 minutes of emergency power from a cell
+	materials = list(MAT_GLASS = 20)
+	rating = 1
+	w_class = WEIGHT_CLASS_TINY
+
+/obj/item/stock_parts/cell/emergency_light/Initialize()
+	. = ..()
+	var/area/A = get_area(loc)

```

Comment:
Does that get the same area as it would for a location object? I wasn't sure if it would possibly return nullspace or somesuch if it was inside of an object.

Reply:
yeah, it's recursive like get_turf

Reply:
Handy!